### PR TITLE
[in_app_purchase] Prepare for `NULLABLE_ARGUMENT_TO_NON_NULL_TYPE`

### PR DESCRIPTION
**Pull request by asashour**
https://github.com/flutter/flutter/issues/121318

This PR modifies the code in order not to trigger the incoming diagnostic.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [X] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [X] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by Hixie** [comment]
test-exempt: fixing lint

This is going to be tricky to land though because it has to be lint-free on both the main branch _and_ the last stable. Search for `_ambiguate` for ways we've handled that before.

File: packages/in_app_purchase/in_app_purchase_storekit/test/fakes/fake_storekit_platform.dart [hunk 1]
```diff
@@ -163,7 +163,9 @@ class FakeStoreKitPlatform {
         return Future<void>.sync(() {});
       case '-[InAppPurchasePlugin retrieveReceiptData:result:]':
         if (receiptData != null) {
-          return Future<String>.value(receiptData);
+          // TODO(asashour): Remove the `ignore` when Dart 3 is released.
```
Comment (stuartmorgan) [inline packages/in_app_purchase/in_app_purchase_storekit/test/fakes/fake_storekit_platform.dart:166]: This should be "reaches stable" not "is released".

**Review by stuartmorgan — APPROVED** [review]
LGTM other than the comment wording

**Comment by stuartmorgan** [comment]
@bparrishMines or @tarrinneal could you do the second review here?

**Review by tarrinneal — APPROVED** [review]

**Pull request merged**